tools/wptrunner/wptrunner/browsers/base.py

```diff
@@ -275,3 +280,123 @@ def __call__(self, line):
         self.logger.process_output(self.pid,
                                    line.decode("utf8", "replace"),
                                    command=" ".join(self.command) if self.command else "")
+
+
+class WebDriverBrowser(Browser):
+    __metaclass__ = ABCMeta
+
+    default_base_path = "/"
+
+    def __init__(self, logger, binary, webdriver_binary, webdriver_args=None,
+                 host="127.0.0.1", port=None, base_path="", env=None, **kwargs):
+        super().__init__(logger)
+
+        if webdriver_binary is None:
+            raise ValueError("WebDriver server binary must be given "
+                             "to --webdriver-binary argument")
+
+        self.logger = logger
+        self.binary = binary
+        self.webdriver_binary = webdriver_binary
+
+        self.host = host
+        self._port = port
+
+        if base_path == "":
+            self.base_path = self.default_base_path
+        else:
+            self.base_path = base_path
+        self.env = os.environ.copy() if env is None else env
+        self._webdriver_args = webdriver_args if webdriver_args is not None else []
```

Comment:
Is there any reason for the underscore here? Why not `self.webdriver_args`?

Reply:
The underscore in this situation is pythonic syntax for a `private` instance variable like you might see in other object-oriented languages. Since the functionality for private variables does not exist in the language, this is the way that devs flag an instance variable as private. No actual functionality changes - just for readability!

Reply:
Sure, but that was not my actual point. Sorry that it was unclear formulated. We have `﻿﻿﻿self.webdriver_binary` on this class and the same style we should use for the arguments. I do not see a specific getter which would require us to use an underscore.

Reply:
I see, I misinterpreted your original comment!